hydra-main/src/main/java/com/addthis/hydra/job/web/resources/JobsResource.java

```diff
@@ -890,21 +893,25 @@ public Response getJobTaskLog(@PathParam("jobID") String jobID,
                 body.put("error", "Missing required query parameter 'node'");
                 return Response.status(Response.Status.BAD_REQUEST).entity(body.toString()).build();
             } else if (port == null) {
-                body.put("error", "Missing required query parameter 'node'");
+                body.put("error", "Missing required query parameter 'port'");
                 return Response.status(Response.Status.BAD_REQUEST).entity(body.toString()).build();
             } else {
-                String url = String.format("http://%s:%s/job.log?id=%s&node=%s&runsAgo=%d&lines=%d&out=%d",
-                        minion, port, jobID, node, runsAgo, lines, out);
+                URI uri = UriBuilder.fromUri("http://" + minion + ":" + port + "/job.log")
+                        .queryParam("id", jobID)
+                        .queryParam("node", node)
+                        .queryParam("runsAgo", runsAgo)
+                        .queryParam("lines", lines)
+                        .queryParam("out", out)
+                        .queryParam("offset", offset)
+                        .build();
+
                 CloseableHttpClient httpclient = HttpClients.createDefault();
```

Comment:
You should reuse the same instance of `HttpClient`, not creating one for each request. See:

http://hc.apache.org/httpcomponents-client-4.5.x/tutorial/html/fundamentals.html#d5e206